### DISCUSSION TITLE
Is it a good Idea to control monthly rate limit with redis?

### DISCUSSION THREAD

**Discussion by softmarshmallow** [nfriedly/express-rate-limit#341] [Q&A]
I am running my service with this package & redis.
For anonymous requests I am using hourly windowms for rate limiting and my redis instance is not using so much mems.

If I start using the same technique for monthly rate limiting, as the keys increase, Is it still a good Idea to use redis for this?
* Since redis is super expensive.

Or will it not take so much mems since its a simple `key (50chars max) : Integer` mapping?

**Comment by nfriedly** [comment]
It probably won't take up all that much memory, but I see a couple of other problems:
* A monthly window means you're really looking for more of a total usage limit / quota rather than a rate-limit. That's not what this project is designed for, so you may run into unforeseen issues trying to use it that way.
* The window for express-rate-limit has to be a fixed size, whereas calendar months vary in length (28-31 days). You could make it 30 days, but you couldn't make it reset on, for example, the 1st of each month.
